demos/face_recognition_demo/python/face_recognition_demo.py

```diff
@@ -0,0 +1,313 @@
+                    (rois[i].position[1] + rois[i].size[1] > orig_image.shape[0]):
+                    continue
+                crop_image = crop(orig_image, rois[i])
+                name = self.faces_database.ask_to_save(crop_image)
+                if name:
+                    id = self.faces_database.dump_faces(crop_image, face_identities[i].descriptor, name)
+                    face_identities[i].id = id
+
+        return [rois, landmarks, face_identities]
+
+    def get_performance_stats(self):
+        stats = {
+            'face_detector': self.face_detector.get_performance_stats(),
+            'landmarks': self.landmarks_detector.get_performance_stats(),
+            'face_identifier': self.face_identifier.get_performance_stats(),
+        }
+        return stats
+
+
+def draw_detections(frame, frame_processor, detections, output_transform):
+    size = frame.shape[:2]
+    frame = output_transform.resize(frame)
+    for roi, landmarks, identity in zip(*detections):
+        text = frame_processor.face_identifier.get_identity_label(identity.id)
+        if identity.id != FaceIdentifier.UNKNOWN_ID:
+            text += ' %.2f%%' % (100.0 * (1 - identity.distance))
+
+        xmin = max(int(roi.position[0]), 0)
+        ymin = max(int(roi.position[1]), 0)
+        xmax = min(int(roi.position[0] + roi.size[0]), size[1])
+        ymax = min(int(roi.position[1] + roi.size[1]), size[0])
+        xmin, ymin, xmax, ymax = output_transform.scale([xmin, ymin, xmax, ymax])
+        cv2.rectangle(frame, (xmin, ymin), (xmax, ymax), (0, 220, 0), 2)
+
+        for point in landmarks:
+            x = xmin + output_transform.scale(roi.size[0] * point[0])
+            y = ymin + output_transform.scale(roi.size[1] * point[1])
+            cv2.circle(frame, (int(x), int(y)), 1, (0, 255, 255), 2)
+        textsize = cv2.getTextSize(text, cv2.FONT_HERSHEY_SIMPLEX, 0.7, 1)[0]
+        cv2.rectangle(frame, (xmin, ymin), (xmin + textsize[0], ymin - textsize[1]), (255, 255, 255), cv2.FILLED)
+        cv2.putText(frame, text, (xmin, ymin), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (0, 0, 0), 1)
+
+    return frame
+
+def center_crop(frame, crop_size):
+    fh, fw, _ = frame.shape
+    crop_size[0], crop_size[1] = min(fw, crop_size[0]), min(fh, crop_size[1])
+    return frame[(fh - crop_size[1]) // 2 : (fh + crop_size[1]) // 2,
+                 (fw - crop_size[0]) // 2 : (fw + crop_size[0]) // 2,
+                 :]
+
+def main():
+    args = build_argparser().parse_args()
+
+    cap = open_images_capture(args.input, args.loop)
+    frame_processor = FrameProcessor(args)
+
+    log.info('Starting inference...')
+    print("To close the application, press 'CTRL+C' here or switch to the output window and press ESC key")
+
+    frame_num = 0
+    metrics = PerformanceMetrics()
+    print_perf_stats = args.perf_stats
+    presenter = None
+    output_transform = None
+    input_crop = None
+    if args.crop_width and args.crop_height:
```

Comment:
Does demo really require cropping input stream? If so, why both keys should be specified? We already have `type=resolution` for similar case.
Also, they are tested for positivity nowhere.

Reply:
I decided to not remove the input cropping from demo, this feature might be useful in some cases. Modified as `--crop_size` key.